### PR TITLE
add __traits getBitfieldOffset and getBitfieldWidth

### PR DESCRIPTION
**Pull request by WalterBright**
Address remaining issues with bit field support.

### CODE REVIEW

**Review by thewilsonator — DISMISSED** [review]
Needs changelog and spec PR

File: compiler/src/dmd/id.d [hunk 1]
```diff
@@ -484,6 +484,8 @@ immutable Msgtable[] msgtable =
     { "hasMember" },
     { "identifier" },
     { "fullyQualifiedName" },
+    { "getBitfieldOffset" },
+    { "getBitfieldWidth" },
```
Comment (thewilsonator) [inline compiler/src/dmd/id.d:488]: Need to update `frontend.h`
```diff
--- a/home/circleci/dmd/compiler/src/dmd/frontend.h
+++ b/home/circleci/dmd/generated/linux/release/64/frontend.h
@@ -8923,6 +8923,8 @@ struct Id final
     static Identifier* hasMember;
     static Identifier* identifier;
     static Identifier* fullyQualifiedName;
+    static Identifier* getBitfieldOffset;
+    static Identifier* getBitfieldWidth;
     static Identifier* getProtection;
     static Identifier* getVisibility;
     static Identifier* parent;
```

**Comment by WalterBright** [comment]
Spec pull: https://github.com/dlang/dlang.org/pull/3921

File: compiler/src/dmd/id.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/id.d:488]: oops

**Comment by rikkimax** [comment]
I can see why you made the traits work with fields that are not bitfields, however given that is has bitfield in their names, it isn't sitting right with me.

If it isn't a bitfield, it could lead to meta-programming code doing bad things if it doesn't error.

**Comment by WalterBright** [comment]
DIP approval is not necessary for this PR, as it enables introspection of ImportC bitfields, too.

**Comment by WalterBright** [comment]
@rikkimax you are pedantically correct, but these traits are of zero usefulness for people who are not using bitfields. Hence it prioritizes ease of use for people using bitfields to find traits for them.

**Comment by rikkimax** [comment]
Right, the code could be designed to handle bit fields.

But the wrong path could be taken (wrong checks, or assumptions about input that didn't hold ext.), and end up using these.

Instead of erroring and going, hey something isn't right, it'll work and do the wrong thing. It'll be very hard to debug.

**Comment by WalterBright** [comment]
I don't see a scenario where it would be very hard to debug.

**Comment by WalterBright** [comment]
@dkorpel ?

**Review by dkorpel — APPROVED** [review]

**Pull request merged**